packages/preload/src/index.ts

```diff
@@ -310,6 +311,35 @@ function initExtensions(): void {
     return getFreePort(port);
   });
 
+  type logFn = (...data: unknown[]) => void;
+
+  contextBridge.exposeInMainWorld('logs', {
+    startReceiveLogs: (
+      providerId: string,
+      log: logFn,
+      warn: logFn,
+      error: logFn,
+      containerConnectionInfo?: ProviderContainerConnectionInfo,
+    ) => {
+      let context;
+      if (containerConnectionInfo) {
+        context = providerRegistry.getMatchingContainerLifecycleContext(providerId, containerConnectionInfo);
+      } else {
+        context = providerRegistry.getMatchingLifecycleContext(providerId);
```

Comment:
I think you could have only one method, with containerConnectionInfo being optional parameter